modules/gke-cluster-autopilot/variables.tf

```diff
@@ -93,7 +93,11 @@ variable "enable_features" {
     vertical_pod_autoscaling = optional(bool, false)
   })
   default = {
-
+    dns = {
+      provider = "CLOUD_DNS"
```

Comment:
Can we use `optional` here? What other providers are supported? If just one, we need to remove the attribute. If two, provider needs to be a `bool`, if more than two we need validation on the value.